gfx/video_shader_parse.c

```diff
@@ -599,178 +972,160 @@ bool video_shader_write_preset(const char *path,
-            *p = '\0';
-         else
+         for ( id = strtok_r(textures_in_conf, ";", &save); 
+               id; 
+               id = strtok_r(NULL, ";", &save))
          {
-            /* if there's no second ", remove whitespace at the end */
-            p--;
-            while (isspace((unsigned char)*p))
-               *p-- = '\0';
+            /* Get the texture path from the override config */
+            if (config_get_path(override_conf, id, override_texture_path, path_size))
+            {
+               /* Resolve the texture's path relative to the override config */
+               if (!path_is_absolute(override_texture_path))
+                  fill_pathname_resolve_relative(resolved_path, 
+                                                override_conf->path, 
+                                                override_texture_path, 
+                                                PATH_MAX_LENGTH);
+               else
+                  strlcpy(resolved_path, override_texture_path, path_size);
+
+               path_relative_to(tmp_rel, resolved_path, tmp_base, tmp_size);
+               config_set_path(override_conf, id, tmp_rel);
+
+               return_val = 1;
+            }
          }
       }
-      else
+      
+      /* ---------------------------------------------------------------------------------
+       * -------------Update Parameter List to include Override Parameters----------------
+       * --------------------------------------------------------------------------------- */
+
+      /* If there is a 'parameters' entry in the override config we want to add these parameters
+       * to the referenced config if they are not already there */
+      if (config_get_array(override_conf, "parameters", override_parameters, param_size))
       {
-         /* remove whitespace at the end (e.g. carriage return) */
-         char *end = ref_path + strlen(ref_path) - 1;
-         while (isspace((unsigned char)*end))
-            *end-- = '\0';
-      }
+         /* Get the string for the parameters from the root config */
+         char *parameters      = NULL;
+         parameters            = (char*)malloc(param_size);
+         parameters[0]         = '\0';
 
-      if (string_is_empty(ref_path))
-      {
-         free(line);
-         return NULL;
+         /* If there are is no parameters entry in the root config, add one */
+         if (!config_get_array(conf, "parameters", parameters, param_size))
+         {
+            config_set_string(conf, "parameters", "");
+            config_get_array(conf, "parameters", parameters, param_size);
+         }
+
+         /* Step through each parameter in override config */
+         for ( id = strtok_r(override_parameters, ";", &save);
+               id; 
+               id = strtok_r(NULL, ";", &save))
+            {
+               /* If the parameter is not in the root config's parameter list add it */
+               if (!strstr(parameters, id))
```

Comment:
`strstr()` is dodgy here. What if you have a parameter whose name matches the prefix of another parameter?